sopel/config/types.py

```diff
@@ -47,6 +47,13 @@ class StaticSection(object):
 
     This class is intended to be subclassed and customized with added
     attributes containing :class:`BaseValidated`-based objects.
+
+    .. note::
+
+        Subclasses of ``StaticSection`` are named ``PluginnameSection`` by
```

Comment:
Does that make sense?

```suggestion
        Subclasses of ``StaticSection`` are named ``PluginNameSection`` by
```

Reply:
Glad to see you're looking out for the same stuff I do. 😁

I flip-flopped on this one while writing it. On one hand, yes, CamelCase dictates that each word should be capitalized. On the other hand, Sopel encourages all-lowercase plugin names, and usually the name is only one word. Multi-word plugin names tend to end up (or should, at least) in snake_case.

So my thought was that this should illustrate the most common case: a one-word plugin name (like `wikipedia` or `calc`) but in generic form (so `pluginname`). There's probably a clearer way to express that this is a generic form, though, that doesn't violate CamelCase expectations… I'll try to figure one out.